ESP32: Disable Arduino OTA by default

fmuntean opened the pull request:
can be enabled back using the #define FEATURE_ARDUINO_OTA

fmuntean committed:
Disable Arduino OTA by default for ESP32

TD-er commented:
Why should it be disabled by default?

TD-er commented:
The build error:
```
Compiling .pio/build/custom_ESP32_4M316k/src/ESPEasy.ino.cpp.o
/home/travis/build/letscontrolit/ESPEasy/src/ESPEasyStorage.ino:8:29: fatal error: ESP_Partition.h: No such file or directory
```

fmuntean commented:
> Why should it be disabled by default?

To match the ESP8266 also the update page is now working.
If enabled by default in code there is no way to disable it.

fmuntean commented:
> The build error:
> 
> ```
> Compiling .pio/build/custom_ESP32_4M316k/src/ESPEasy.ino.cpp.o
> /home/travis/build/letscontrolit/ESPEasy/src/ESPEasyStorage.ino:8:29: fatal error: ESP_Partition.h: No such file or directory
> ```

Hmm, can I say "It works on my machine?" ;)
I will investigate this further. maybe it depends on the ESP32 framework version.

TD-er commented:
> > Why should it be disabled by default?
> 
> To match the ESP8266 also the update page is now working.
> If enabled by default in code there is no way to disable it.

The update page on ESP32 was working.
So not sure what you changed then to make it work?
Or was it not working when ARDUINO_OTA was disabled?

TD-er commented:
> > The build error:
> > ```
> > Compiling .pio/build/custom_ESP32_4M316k/src/ESPEasy.ino.cpp.o
> > /home/travis/build/letscontrolit/ESPEasy/src/ESPEasyStorage.ino:8:29: fatal error: ESP_Partition.h: No such file or directory
> > ```
> 
> Hmm, can I say "It works on my machine?" ;)
> I will investigate this further. maybe it depends on the ESP32 framework version.

If so, please check what version you're running and from where this file is included.

fmuntean commented:
the ESP_Partition.h is under .platformio\packages\framework-arduinoespressif32\tools\sdk\include\spi_flash
I am able to build : test_ESP32_4M316k                              SUCCESS   00:02:20.620
same for: custom_ESP32_4M316k                            SUCCESS   00:01:56.272

Processing custom_ESP32_4M316k (platform: espressif32@1.11.2; board: esp32dev; framework: arduino)

I added the -D CONTINUOUS_INTEGRATION to the build flags for test_ESP32_4M316k    and the build is failing here:
D:/git/HomeAutomation/ESPEasy/src/ESPEasy.ino: In function 'void backgroundtasks()':
D:/git/HomeAutomation/ESPEasy/src/ESPEasy.ino:961:14: error: unused variable 'wifiConnected' [-Werror=unused-variable]
   const bool wifiConnected = WiFiConnected();
              ^

@TD-er I think I need your help here as I don't know how the CI build works. As that is not something that I changed please let me know if you need anything from me.
From what I can see that variable is only used when board is ESP8266

fmuntean commented:
This is a build log with no verbose. hope it can help finding the issue:
```
Processing test_ESP32_4M316k (platform: espressif32@1.11.2; board: esp32dev; framework: arduino)
------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
 Compile time defines 
 PROGNAME:         ESP_Easy_mega_20200523_test_ESP32_4M316k
 BUILD_PLATFORM:   Windows-10-10.0.18362-SP0
 GIT_HEAD:         ESP32_ARDUINO_OTA
 ------------------------------- 
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 1.11.2 > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 1.81MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 3.10004.200129 (1.0.4)
 - tool-esptoolpy 1.20600.0 (2.6.0)
 - toolchain-xtensa32 2.50200.80 (5.2.0)
Converting ESPEasy.ino
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 64 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <ESPeasySerial> 2.0.3 #737cfaf
|   |-- <SC16IS752> 1.0.1
|   |   |-- <SPI> 1.0
|   |   |-- <Wire> 1.0.1
|-- <Adafruit ILI9341> 1.5.5
|   |-- <Adafruit GFX Library> 1.8.3
|   |   |-- <Adafruit BusIO> 1.2.6
|   |   |   |-- <Wire> 1.0.1
|   |   |   |-- <SPI> 1.0
|   |   |-- <SPI> 1.0
|   |   |-- <Wire> 1.0.1
|   |-- <Adafruit STMPE610> 1.1.2
|   |   |-- <SPI> 1.0
|   |   |-- <Wire> 1.0.1
|   |-- <Adafruit TouchScreen> 1.1.0
|   |-- <SPI> 1.0
|-- <Adafruit GFX Library> 1.8.3
|   |-- <Adafruit BusIO> 1.2.6
|   |   |-- <Wire> 1.0.1
|   |   |-- <SPI> 1.0
|   |-- <SPI> 1.0
|   |-- <Wire> 1.0.1
|-- <SPIFFS> 1.0
|   |-- <FS> 1.0
|-- <DNSServer> 1.1.0
|   |-- <WiFi> 1.0
|-- <ESPmDNS> 1.0
|   |-- <WiFi> 1.0
|-- <FS> 1.0
|-- <I2Cdevlib-Core>
|   |-- <Wire> 1.0.1
|-- <SD(esp32)> 1.0.5
|   |-- <FS> 1.0
|   |-- <SPI> 1.0
|-- <SPI> 1.0
|-- <WiFi> 1.0
|-- <Wire> 1.0.1
|-- <Adafruit SGP30 Sensor> 1.0.5
|   |-- <Wire> 1.0.1
|-- <Adafruit TCS34725> 1.0.1
|   |-- <Wire> 1.0.1
|-- <Adafruit TSL2591 Library> 1.0.2
|   |-- <Adafruit Unified Sensor> 1.0.2
|   |-- <Wire> 1.0.1
|-- <ccronexpr>
|-- <ESP8266_SSD1306> 3.2.7
|   |-- <Wire> 1.0.1
|   |-- <SPI> 1.0
|-- <HTTPClient> 1.2
|   |-- <WiFi> 1.0
|   |-- <WiFiClientSecure> 1.0
|   |   |-- <WiFi> 1.0
|-- <Adafruit Motor Shield V2 Library> 1.0.4
|   |-- <Wire> 1.0.1
|-- <Adafruit MPR121> 1.0.0
|   |-- <Wire> 1.0.1
|-- <Adafruit NeoPixel> 1.0.6
|-- <Adafruit Unified Sensor> 1.0.2
|-- <AM2320>
|   |-- <Wire> 1.0.1
|-- <ArduinoJson> 6.11.0
|-- <Blynk> 0.6.1
|   |-- <WiFiClientSecure> 1.0
|   |   |-- <WiFi> 1.0
|-- <DLBus>
|-- <HLW8012> 1.1.1
|-- <CHT16K33 LED Matrix Library> 0.0.1
|   |-- <Wire> 1.0.1
|-- <SerialSensors> 0.0.1
|   |-- <ESPeasySerial> 2.0.3 #737cfaf
|   |   |-- <SC16IS752> 1.0.1
|   |   |   |-- <SPI> 1.0
|   |   |   |-- <Wire> 1.0.1
|-- <LiquidCrystal_I2C>
|   |-- <Wire> 1.0.1
|-- <NewPingESP8266>
|-- <MechInputs> 0.0.1
|-- <Regexp> 0.1.0
|-- <RN2xx3 Arduino Library> 1.0.1
|-- <SDM> 2.0.0
|   |-- <ESPeasySerial> 2.0.3 #737cfaf
|   |   |-- <SC16IS752> 1.0.1
|   |   |   |-- <SPI> 1.0
|   |   |   |-- <Wire> 1.0.1
|-- <SparkFun APDS9960 RGB and Gesture Sensor> 1.4.2
|   |-- <Wire> 1.0.1
|-- <TinyGPSPlus> 1.0.2
|-- <WebServer> 1.0
|   |-- <WiFi> 1.0
|   |-- <FS> 1.0
|-- <PubSubClient> 2.7
|-- <ArduinoOTA> 1.0
|   |-- <Update> 1.0
|   |-- <WiFi> 1.0
|   |-- <ESPmDNS> 1.0
|   |   |-- <WiFi> 1.0
|-- <ESP32HTTPUpdateServer>
|   |-- <FS> 1.0
|   |-- <Update> 1.0
|   |-- <WebServer> 1.0
|   |   |-- <WiFi> 1.0
|   |   |-- <FS> 1.0
|   |-- <WiFi> 1.0
Building in release mode
Compiling .pio\build\test_ESP32_4M316k\src\ESPEasy.ino.cpp.o
Linking .pio\build\test_ESP32_4M316k\ESP_Easy_mega_20200523_test_ESP32_4M316k.elf
Building .pio\build\test_ESP32_4M316k\ESP_Easy_mega_20200523_test_ESP32_4M316k.bin
Retrieving maximum program size .pio\build\test_ESP32_4M316k\ESP_Easy_mega_20200523_test_ESP32_4M316k.elf
Checking size .pio\build\test_ESP32_4M316k\ESP_Easy_mega_20200523_test_ESP32_4M316k.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [==        ]  16.7% (used 54840 bytes from 327680 bytes)
Flash: [========= ]  92.8% (used 1763466 bytes from 1900544 bytes)
esptool.py v2.6
esp32_create_factory_bin([".pio\build\test_ESP32_4M316k\ESP_Easy_mega_20200523_test_ESP32_4M316k.bin"], [".pio\build\test_ESP32_4M316k\ESP_Easy_mega_20200523_test_ESP32_4M316k.elf"])
Generating factory bin for genuine esp units
bin_elf_copy([".pio\build\test_ESP32_4M316k\ESP_Easy_mega_20200523_test_ESP32_4M316k.bin"], [".pio\build\test_ESP32_4M316k\ESP_Easy_mega_20200523_test_ESP32_4M316k.elf"])
 copy to:   build_output\debug\ESP_Easy_mega_20200523_test_ESP32_4M316k.elf
 copy to:   build_output\bin\ESP_Easy_mega_20200523_test_ESP32_4M316k.bin
 copy to:   build_output\bin\ESP_Easy_mega_20200523_test_ESP32_4M316k-factory.bin
 copy to:   build_output\debug\ESP_Easy_mega_20200523_test_ESP32_4M316k.env.txt
=================================================== [SUCCESS] Took 72.94 seconds ===================================================
```

fmuntean commented:
Alo if I do the following inside the EsPEasy.ino:
START_TIMER
  #if defined(ESP8266)
  const bool wifiConnected = WiFiConnected();
  #else
  WiFiConnected();  
  #endif

Then the build is successful when using the -D CONTINUOUS_INTEGRATION
From what I can see the CI should have failed even before this change.

fmuntean commented:
@TD-er : any update on this ?

TD-er commented:
> @TD-er : any update on this ?

Not really.
Yesterday I noticed we're running behind a bit in the ESP32 code, so I tried building for ESP32 using the last PlatformIO tag for ESP32.
But it stil doesn't use the latest code from the ESP32 Arduino repo. (I tried to use the WiFi scan function to scan per channel and that cannot build, but code merged a few months ago was not yet present)

So that's how far I got.

TD-er committed:
Merge branch 'mega' into ESP32_ARDUINO_OTA

fmuntean commented:
Any update on this? Anything I can do to help move this forward?

TD-er committed:
Merge branch 'mega' into ESP32_ARDUINO_OTA

TD-er commented:
> Any update on this? Anything I can do to help move this forward?

Just merged in a fix I already had for it in the mega branch.

fmuntean commented:
> > Any update on this? Anything I can do to help move this forward?
> 
> Just merged in a fix I already had for it in the mega branch.

Thanks:
@TD-er : However from what I can see it is still enabled by default for ESP32 where is disabled for ESP8266 inside the ESP-Globals.h line 117

fmuntean commented:
@TD-er : From what I can see it is still enabled by default for ESP32 inside the ESP-Globals.h line 117
Am I missing something here?

TD-er committed:
Merge branch 'mega' into ESP32_ARDUINO_OTA

TD-er commented:
I will now have a look at it.

TD-er committed:
[ESP32] Only define FEATURE_ARDUINO_OTA in build config, not in .h files

TD-er committed:
Merge remote-tracking branch 'fmuntean/ESP32_ARDUINO_OTA' into ESP32_ARDUINO_OTA
